app/decorators/member_decorator.rb

```diff
@@ -0,0 +1,21 @@
+class MemberDecorator < Draper::Decorator
+  delegate_all
+
+  def worked_hours_of_three_last_months
+    hours_per_month = []
+    3.times do |n|
+      hours_per_month[n] = h.content_tag(:p, "#{I18n.localize(Date.current - n.month, format: :only_month)} :
+                                              #{model.monthly_worked_hours(Date.current - n.month)}")
+    end
+    h.safe_join hours_per_month.reverse
+  end
+
+  def worked_hours_of_three_last_months_csv_version
```

Comment:
Since those two methods are nearly the same, you could refactor to combine them together, by giving a `csv: true` keyword argument to signify you want the csv version. E.g.: `def hours_worked_in_the_last_three_months(csv: true)`
You could then separate the steps according to that option.
Also, instead of declaring `hours_per_month = []` before the `each` loop, you can try the ruby method `.with_object`

For example:
```ruby
def hours_worked_in_last_three_months(csv: false)
  hours_per_month = 3.times.with_object([]) do |n, array|
                      array[n] = "#{I18n.localize(Date.current - n.month, format: :only_month)} :
                                  #{model.monthly_worked_hours(Date.current - n.month)}"
                    end
  if csv
    hours_per_month.reverse.join("\n")
  else
    h.safe_join(hours_per_month.map { |month_total| h.content_tag(:p, month_total) }.reverse)
  end
end
```

This is just an example as a suggestion. I suppose it should work but I don't know. If you do this refactor, write some tests to validate the behaviour

Reply:
done